### ISSUE TITLE
Constructor-like functions are lost from functions list if they have the same decapitalized analog

### ISSUE THREAD

**Issue by sandwwraith** [Kotlin/dokka#2950]
**Describe the bug**
Sounds a bit hard to understand, so I'll show the example. In kotlinx.serialization, there is:

`interface SerialDescriptor` — https://github.com/Kotlin/kotlinx.serialization/blob/60c632c970d7da02151fd7c920f40ebc269c0f29/core/commonMain/src/kotlinx/serialization/descriptors/SerialDescriptor.kt#L146

`public fun SerialDescriptor` — https://github.com/Kotlin/kotlinx.serialization/blob/ccf9c2c52cd9df10e3782f9f8a02ff4172c4b868/core/commonMain/src/kotlinx/serialization/descriptors/SerialDescriptors.kt#L116

`public inline fun <reified T> serialDescriptor()`(2 overloads) — https://github.com/Kotlin/kotlinx.serialization/blob/ccf9c2c52cd9df10e3782f9f8a02ff4172c4b868/core/commonMain/src/kotlinx/serialization/descriptors/SerialDescriptors.kt#L153

However, on the https://kotlinlang.org/api/kotlinx.serialization/kotlinx-serialization-core/kotlinx.serialization.descriptors/ page we can see only `interface SerialDescriptor` on Types tab, and Functions tab with lowercase names (despite function itself in the table named as `SerialDescriptor`:

<img width="1162" alt="image" src="https://user-images.githubusercontent.com/11777575/228915261-fabcc59f-d550-4a43-b9ed-71e926fdb0e4.png">

There is, however, page for `fun SerialDescriptor`: https://kotlinlang.org/api/kotlinx.serialization/kotlinx-serialization-core/kotlinx.serialization.descriptors/-serial-descriptor.html 
and it is clickable from sidebar:

<img width="256" alt="image" src="https://user-images.githubusercontent.com/11777575/228915543-40c6e67a-27e2-4906-9c79-635ea2990858.png">

**Expected behaviour**
`fun SerialDescriptor` is also listed on the Functions tab

**Installation**
- Dokka version: 1.7.0

**Additional context**
Website is build on teamcity

**Comment by IgnatBeresnev** [comment]
This might've been "fixed" in the recent versions of Dokka, either in #2846 or in #2861. 

This is how it currently looks if you open the "Functions" tab of the package, would this behaviour be acceptable?

![2023-04-06_22-18-57](https://user-images.githubusercontent.com/22685910/230485278-ff4aafc3-f35a-4a50-b793-118fa067fc74.png)

**Comment by sandwwraith** [comment]
Yes, this looks OK. Is this a build on your machine? What version of Dokka should I update to?

**Comment by IgnatBeresnev** [comment]
@sandwwraith the fixes should be present in Dokka 1.8.10

I'll close the issue once Dokka is updated in kotlinx.serialization, tagging it as "awaiting response" to not forget

**Comment by IgnatBeresnev** [comment]
@sandwwraith the API reference for serialization has been updated, can you please have a look and say if the issue is still relevant?

**Comment by sandwwraith** [comment]
It has all functions now, thanks!
